golos.publication/golos.publication.cpp

```diff
@@ -124,14 +125,34 @@ void publication::create_message(name account, std::string permlink,
     }
     eosio_assert((benefic_map.size() <= max_beneficiaries_param.max_beneficiaries), "publication::create_message: benafic_map.size() > MAX_BENEFICIARIES");
 
+    auto obj_referral = golos::referral::account_referrer( config::referral_name, account );
```

Comment:
These actions must be performed if the parameters of the posting contract specify the referral program contract (see https://github.com/GolosChain/golos-smart/pull/383). I still can’t merge those changes, but you can rebase your PR on them and use the added parameter of the referral program.